### PR TITLE
Add implementation details to tokenized auth

### PR DESCRIPTION
**Pull request by jsafrane**
Add env. variable + annotation name to the proposal, so operator authors don't need to read a separate godoc that's circling around. I am just reflecting current implementation.

And mark the enhancement as implementable, it has been already implemented.

### CODE REVIEW

**Comment by jsafrane** [comment]
I am just reflecting current implementation in the enhancement.
cc original authors: @bentito @jharrington22 @gallettilance
approvers: @sdodson @joelanford @bparees
API approver (annotation is API :-) ): @JoelSpeed

File: enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md [hunk 1]
```diff
@@ -208,11 +210,20 @@ enabled cluster. New guidelines would include the following to use CCO has detec
...
+            features.operators.openshift.io/token-auth-aws: "true"
+            ...
+    ```
 - Add a script in the operator description, per supported cloud provider, to help set up the correct role
 - (Optional) Add the projected ServiceAccount volume to the Deployment embedded in the CSV;
-- handle the Secret themselves (like today’s CCO Mint mode, read and use the credentials in the Secret whose name they 
-  know from the CredentialsRequest and is in the same Namespace)
+- When OLM starts the operator with `ROLEARN` env. variable, handle the Secret themselves (like today’s CCO Mint mode,
+  read and use the credentials in the Secret whose name they know from the CredentialsRequest and is in the same
+  Namespace.)
```
Comment (bparees) [inline enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md:226]: aren't we missing the step where the operator is expected to create the CredReq object, also, when the cluster is in STS mode?

Comment (jsafrane) [reply enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md]: I assume that when `ROLEARN` is set, then OLM / console has checked CCO is in STS mode already.

I rephrased the operator behavior, now it should be clearer.

**Comment by JoelSpeed** [comment]
Annotation based APIs are generally a no, but in this case there is prior art and good reason (can't add a new field for every OLM operator that needs to indicate something to the cluster), so we allow annotations in this particular case

No further feedback from me, will allow the other approvers to chime in for tags

**Comment by bparees** [comment]
/approve

@gallettilance any other details from https://docs.google.com/document/d/1iFNpyycby_rOY1wUew-yl3uPWlE00krTgr9XHDZOTNo/edit# that we should pull in here?

File: enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md [hunk 2]
```diff
@@ -208,12 +210,22 @@ enabled cluster. New guidelines would include the following to use CCO has detec
...
+            ...
+    ```
 - Add a script in the operator description, per supported cloud provider, to help set up the correct role
 - (Optional) Add the projected ServiceAccount volume to the Deployment embedded in the CSV;
-- handle the Secret themselves (like today’s CCO Mint mode, read and use the credentials in the Secret whose name they 
-  know from the CredentialsRequest and is in the same Namespace)
-- Add structure to controller code to be resilient to missing cloud credentials / not crash.
+- When OLM starts the operator with `ROLEARN` env. variable, handle the cloud credentials *almost* as if CCO is in Mint
+  mode:
+  - Create CredentialsRequest, now with `spec.cloudTokenPath` and `spec.providerSpec.stsIAMRoleARN` filed.
```
Comment (gallettilance) [inline enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md:226]:
```suggestion
  - Create CredentialsRequest, now with `spec.cloudTokenPath` and `spec.providerSpec.stsIAMRoleARN` filled.
```

File: enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md [hunk 3]
```diff
@@ -208,12 +210,22 @@ enabled cluster. New guidelines would include the following to use CCO has detec
...
+  - Wait for Secrets referenced in the CredentialsRequest to be created by CCO.
```
Comment (gallettilance) [inline enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md:227]:
```suggestion
  - Wait for the Secret referenced in the CredentialsRequest to be created by CCO.
```

File: enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md [hunk 4]
```diff
@@ -208,12 +210,22 @@ enabled cluster. New guidelines would include the following to use CCO has detec
...
+    - Expect the Secrets take some time to create, do not crash and report error when it takes too long.
```
Comment (gallettilance) [inline enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md:228]:
```suggestion
    - Expect the Secret to take some time to create, do not crash and report an error when it takes too long.
```

**Review by gallettilance — COMMENTED** [review]
Just some nits. It'd be good to also include the case where the env var is modified in the subscription (which will restart the pod) and it's the operator's responsibility to check that the already created credReq's IAMRole matches the env var.

**Review by gallettilance — APPROVED** [review]
/lgtm

**Pull request merged**